src/pybind/mgr/cephadm/migrations.py

```diff
@@ -442,6 +447,72 @@ def migrate_6_7(self) -> bool:
         # was set to true. Therefore we have nothing to migrate for those daemons
         return True
 
+    def migrate_7_8(self) -> bool:
+        """
+        Migration 7 -> 8
+        Replace Promtail with Alloy.
+
+        For this migration:
+        - If mgr daemons are still being upgraded, return True WITHOUT bumping migration_current.
+        (avoids failover to an old mgr that doesn't know Alloy).
+        - Once all mgrs are upgraded, remove all Promtail daemons + service.
+        - Only after confirming Promtail is gone, deploy Alloy on those hosts.
+        """
+        if not self.all_mgrs_upgraded():
+            logger.info("Promtail→Alloy migration: mgr daemons still upgrading. "
```

Comment:
I'm old school in that I like to keep logs and messages 7-bit-clean, thus I would suggest replacing → with `->` here and on various lines below